src/ovs/ovs_control.cpp

```diff
@@ -395,11 +398,101 @@ OVS_Control::set_protocol_for_flow_dump(vconn *vconn,
         ovs_fatal(0, "none of the usable flow formats (%s) is among the "
                   "allowed flow formats (%s)", usable_s, allowed_s);
     }
+    return (ofputil_protocol) 0;
+}
+
+void
+OVS_Control::add_flow(const char *bridge, const char *flow)
+{
+    flow_mod(bridge, flow, OFPFC_ADD);
+}
+
+void
+OVS_Control::mod_flows(const char *bridge, const char *flow, bool strict)
+{
+    flow_mod(bridge, flow, strict ? OFPFC_MODIFY_STRICT : OFPFC_MODIFY);
+}
+
+void
+OVS_Control::del_flows(const char *bridge, const char *flow, bool strict)
+{
+    flow_mod(bridge, flow, strict ? OFPFC_DELETE_STRICT : OFPFC_DELETE);
+}
+
+void
+OVS_Control::flow_mod(const char *bridge, const char *flow, unsigned short int command)
+{
+    struct ofputil_flow_mod fm;
+    char *error;
+    enum ofputil_protocol usable_protocols;
+    
+    error = parse_ofp_flow_mod_str(&fm, flow,
+                                    ports_to_accept(bridge),
+                                    tables_to_accept(bridge), command,
+                                    &usable_protocols);
+    if (error) {
+        ovs_fatal(0, "%s", error);
```

Comment:
what will ovs_fatal do? will it throw exception or exit ACA? If yes, we will need to handle it better.